logllm/log_llm.py

```diff
@@ -1,44 +1,79 @@
-from openai import OpenAI
+import google.generativeai as genai
 import wandb
-from .extractor import extract_notebook_code
+from logllm import extract_notebook_code
 import json
+import os
+from dotenv import load_dotenv
+import markdown as md
+from openai import OpenAI
+
+# Load environment variables
+load_dotenv()
 
+# Configure Google Generative AI
+genai.configure(api_key=os.getenv('API_KEY'))
+
+generation_config = {
+    "temperature": 0,
+    "top_p": 0.95,
+    "top_k": 64,
+    "max_output_tokens": 8192,
+    "response_mime_type": "application/json",
+}
+
+model = genai.GenerativeModel(
+    model_name="gemini-1.5-flash",
+    generation_config=generation_config,
+)
 
 def init_wandb(project_name):
     wandb.init(project=project_name, settings=wandb.Settings(_disable_stats=True))
 
-def extract_experimental_conditions(code):
+def extract_experimental_conditions_gemini(code):
```

Comment:
I think you do not have to make two functions, you can have one function where you use if branch to switch openai and gemini